src/core/filesystem.cc

```diff
@@ -1283,22 +1349,27 @@ S3FileSystem::S3FileSystem(
   const char* key_id = std::getenv("AWS_ACCESS_KEY_ID");
   const char* region = std::getenv("AWS_DEFAULT_REGION");
   const char* session_token = std::getenv("AWS_SESSION_TOKEN");
+  const char* profile_name = std::getenv("AWS_PROFILE");
   if ((secret_key != NULL) && (key_id != NULL)) {
     credentials.SetAWSAccessKeyId(key_id);
     credentials.SetAWSSecretKey(secret_key);
     if (session_token != NULL) {
       credentials.SetSessionToken(session_token);
     }
     config = Aws::Client::ClientConfiguration();
-    if (region != NULL) {
-      config.region = region;
-    }
-  } else if (const char* profile_name = std::getenv("AWS_PROFILE")) {
+  } else if (profile_name != NULL) {
     config = Aws::Client::ClientConfiguration(profile_name);
   } else {
     config = Aws::Client::ClientConfiguration("default");
   }
 
+  if (region != NULL) {
+    config.region = region;
+  }
+
+  config.connectTimeoutMs = 30000;
```

Comment:
Why are you hardcoding these values and is thus necessary?

Reply:
Nope. I will fix it.